exchanges/subscription/keys.go

```diff
@@ -0,0 +1,62 @@
+package subscription
+
+// MatchableKey interface should be implemented by Key types which want a more complex matching than a simple key equality check
+// The Subscription method allows keys to compare against keys of other types
+type MatchableKey interface {
+	Match(MatchableKey) bool
+	GetSubscription() *Subscription
+}
+
+// ExactKey is key type for subscriptions where all the pairs in a Subscription must match exactly
+type ExactKey struct {
+	*Subscription
+}
+
+var _ MatchableKey = ExactKey{} // Enforce ExactKey must implement MatchableKey
+
+// GetSubscription returns the underlying subscription
+func (k ExactKey) GetSubscription() *Subscription {
+	return k.Subscription
+}
+
+// Match implements MatchableKey
+// Returns true if the key fields exactly matches the subscription, including all Pairs
+func (k ExactKey) Match(eachKey MatchableKey) bool {
+	eachSub := eachKey.GetSubscription()
```

Comment:
nil check on `eachKey` && `eachsub`

Reply:
Yeah, good point. If eachSub came out nil then we'd panic below.

Fixed bec4472